File: submissions/project-euler/13.py
'''
faktor = [0,1]

for i in range(2,500000):
    ans = 2
    for j in range(2,i+1):
        if(j*j > i):
            break
        else:
            if(i%j == 0):
                if(j*j==i):
                    ans += 1
                else:
                    ans += 2
    faktor.append(ans)

def countFactorTriangular(x):
    if x%2==0:
        return faktor[x//2]*faktor[x+1]
    else:
        return faktor[(x+1)//2]*faktor[x]

t = int(input("Masukkan nilai: "))
n = 1

while(countFactorTriangular(n)<=500):
    n+=1

'''
print(12375*(12376)/2)

File: submissions/online-judges/project-euler/13.cpp
#include<bits/stdc++.h>
using namespace std;

int dp[]

int countFactor(){
    int ans = 2;
    for(int j=2; j*j<=x; j++){
        if(x%j==0){
            if(j*j==x) ans++;
            else ans += 2;
        }
    }
    return ans;
}

int countFactorTriangular(int n){
    if(n%2==0){
        return countFactor(n/2) + countFactor(n+1);
    } else {
        return countFactor((n+1)/2) + countFactor(n);
    }
}

int main(){
    int n=1, k;
    cout << "Masukkan nilai: ";
    cin >> k;
    while(countFactorTriangular(n) <= k){
        n++;
    }
    cout << n;
}